src/saved_game.cpp

```diff
@@ -296,47 +296,40 @@ void saved_game::expand_scenario()
 
 void saved_game::check_require_scenario()
 {
-	if(starting_point_["addon_id"].empty()) {
-		return;
-	}
-
-	config required_scenario;
-
-	required_scenario["id"] = starting_point_["addon_id"];
-	required_scenario["name"] = starting_point_["addon_title"];
-	required_scenario["version"] = starting_point_["addon_version"];
-	required_scenario["min_version"] = starting_point_["addon_min_version"];
-	required_scenario["required"] = starting_point_["require_scenario"].to_bool(false);
-	config& content = required_scenario.add_child("content");
+	config scenario;
+	scenario["id"] = starting_point_["addon_id"].str("mainline");
+	scenario["name"] = starting_point_["addon_title"].str("mainline");
+	scenario["version"] = starting_point_["addon_version"];
+	scenario["min_version"] = starting_point_["addon_min_version"];
+	scenario["required"] = starting_point_["require_scenario"].to_bool(false);
+	config& content = scenario.add_child("content");
 	content["id"] = starting_point_["id"];
 	content["type"] = "scenario";
 
-	mp_settings_.update_addon_requirements(required_scenario);
+	mp_settings_.update_addon_requirements(scenario);
 }
 
-void saved_game::load_mod(const std::string& type, const std::string& id, size_t pos)
+void saved_game::load_non_scenario(const std::string& type, const std::string& id, size_t pos)
```

Comment:
I think this should have a better name or at least a short description what a "non-scenario" is. (campaign/era/modification/resource/more?)

Reply:
Currently it's loading the mods, era, campaign, and resources.  I'm not sure what a better name would be, though I think `load_non_scenario` is more accurate than `load_mod`.

Reply:
So a description could be: "non-scenario" here means modification, era, campaign or resource.

Anyway I've approved since these are just minor suggestions. Feel free to disregard if you don't think they're useful.

Reply:
I'll add it as a comment then.